kem/kyber/templates/pkg.templ.go

```diff
@@ -256,19 +256,41 @@ func (sk *PrivateKey) Pack(buf []byte) {
 // Unpacks sk from buf.
 //
 // Panics if buf is not of size PrivateKeySize.
+{{ if .NIST -}}
+//
+// Returns an error if buf is not of size PrivateKeySize, or private key
+// doesn't pass the ML-KEM decapsulation key check.
+func (sk *PrivateKey) Unpack(buf []byte) error {
+	if len(buf) != PrivateKeySize {
```

Comment:
With this addition, the extra check in `UnmarshalBinaryPrivateKey` for the same condition becomes redundant. Can we simplify the implementation template by unconditionally including this check and returning `error` for this function?

Are there users of `Unpack` other than `UnmarshalBinaryPrivateKey`? If so, should we try harder to prevent intermediate values in `sk.sk`, `sk.pk`, `sk.hpk` and `sk.z` when the error check fails? (E.g. by writing these only at the end, or clearing values on error?)

Reply:
I don't want to change the existing Kyber API.